repair/featurize/freqfeat.py

```diff
@@ -5,16 +5,13 @@
 
 
 class FreqFeaturizer(Featurizer):
-    def __init__(self, name='FreqFeaturizer'):
-        super(FreqFeaturizer, self).__init__(name)
-
     def specific_setup(self):
+        self.name = 'FreqFeaturizer'
+        self.all_attrs = self.ds.get_attributes()
         self.attrs_number = len(self.ds.attr_to_idx)
         total, single_stats, pair_stats = self.ds.get_statistics()
         self.total = total
-        self.single_stats = {}
-        for attr in single_stats:
-            self.single_stats[attr] = single_stats[attr].to_dict()
+        self.single_stats = single_stats
 
     def gen_feat_tensor(self, input, classes):
```

Comment:
gen_feat_tensor is sometimes an instance method in some featurizers and a normal function outside of the class in others.